### DISCUSSION TITLE
Table Visualizations: Download as CSV includes hidden columns

### DISCUSSION THREAD

**Discussion by loreburek** [getredash/redash#6126] [Q&A]
hi! 
I have a Query that pulls data for a data transfer, plus some internal fields that are used for other tasks.  I have a Table Visualization set up that hides the internal fields, but the Download as CSV continues to download the full query results.

To replicate: 
1. Create a new query with more fields than you want to download
`select a, b, c,d from t_table t
limit 10`
2. Save and Execute the Query.
3. In the resulting Table Visualization, click the kebab menu and select [Download as CSV]. 
4. Your CSV contains the complete queryset with 4 columns. 
5. In the same table visualization, click [Edit Visualization], and uncheck the column you'd like to exclude.  Save, of course. 
6. Kebab --> [Download as CSV] --> all fields are still included.

**Comment by gaecoli** [comment]
At present, the exported file is exported completely, and there is no filtering function.

**Comment by bmoygh** [comment]
Does anyone know if there is an update to this?

**Reply by wrightling** [reply]
I'm looking for this as well - would it be useful to submit this request as a feature request instead of Q&A discussion? I saw that some active development started up this year on redash, which is exciting.
